api/src/paths/project/{projectId}/survey/{surveyId}/critters/{critterId}/telemetry.ts

```diff
@@ -0,0 +1,308 @@
+                              type: 'boolean'
+                            },
+                            map_colour: {
+                              type: 'string'
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              },
+              tracks: {
+                type: 'object',
+                properties: {
+                  type: {
+                    type: 'string'
+                  },
+                  geometry: {
+                    type: 'object',
+                    properties: {
+                      type: {
+                        type: 'string'
+                      },
+                      coordinates: {
+                        type: 'array',
+                        items: {
+                          type: 'array',
+                          items: {
+                            type: 'number'
+                          }
+                        }
+                      }
+                    }
+                  },
+                  properties: {
+                    type: 'object',
+                    properties: {
+                      critter_id: {
+                        type: 'string',
+                        format: 'uuid'
+                      },
+                      deployment_id: {
+                        type: 'string',
+                        format: 'uuid'
+                      },
+                      map_colour: {
+                        type: 'string'
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function getCritterTelemetry(): RequestHandler {
+  return async (req, res) => {
+    const user: ICritterbaseUser = {
+      keycloak_guid: req['system_user']?.user_guid,
+      username: req['system_user']?.user_identifier
+    };
+    const critterId = Number(req.params.critterId);
+    const surveyId = Number(req.params.surveyId);
+    const startDate = String(req.query.startDate);
+    const endDate = String(req.query.endDate);
+    const connection = getDBConnection(req['keycloak_token']);
+    const surveyCritterService = new SurveyCritterService(connection);
+    const bctw = new BctwService(user);
+    try {
+      await connection.open();
+      const surveyCritters = await surveyCritterService.getCrittersInSurvey(surveyId);
+      const thisCritter = surveyCritters.find((a) => a.critter_id === critterId);
+      if (!thisCritter) {
+        throw Error('Specified critter was not part of this survey.');
```

Comment:
Should we be throwing HTTP errors here instead? I know the BCTW endpoints work a little differently